langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Sort.java

```diff
@@ -42,12 +48,16 @@
 public class Sort {
 
     public static BArray sort(BArray arr, Object direction, Object func) {
-        checkIsArrayOnlyOperation(TypeUtils.getImpliedType(arr.getType()), "sort()");
+        BArray sortedArray;
+        Type arrType = arr.getType();
         BFunctionPointer<Object, Object> function = (BFunctionPointer<Object, Object>) func;
-
+        // Check if the array type is an Ordered type, otherwise a key function is mandatory
+        if (!isOrderedType(arrType) && function == null) {
+            throw ErrorCreator.createError(getModulePrefixedReason(ARRAY_LANG_LIB, INVALID_TYPE_TO_SORT),
+                    StringUtils.fromString("Valid key function required"));
```

Comment:
Shall we start with lower case to comply with other Ballerina error messages. Check `MessagesBundle.properties`

Reply:
Fixed in 47f707437372939e6864383c24e81e2afb5f2500